### ISSUE TITLE
lr decay with gradient accumulation

### ISSUE THREAD

**Issue by khoa-ho** [IndicoDataSolutions/finetune#380]
**Describe the bug**
I didn't seem to get the right behavior with gradient accumulation, shouldn't this https://github.com/IndicoDataSolutions/finetune/blob/development/finetune/model.py#L201 includes `params.accum_steps` like this `total_num_steps = params.n_epochs * params.dataset_size // (params.batch_size * params.accum_steps * n_replicas)`

**Comment by madisonmay** [comment]
Good call! @benleetownsend can you confirm?

**Comment by madisonmay** [comment]
After some debugging / logging it appears the `lr_decay` is computed properly without division by the factor of `params.accum_steps`.  See https://github.com/IndicoDataSolutions/finetune/blob/development/finetune/optimizers/gradient_accumulation.py for details as to why.  

Were you logging lr_decay or was it something else that made you suspect a problem with gradient accumulation?

**Comment by khoa-ho** [comment]
Actually the weird behavior I saw with lr warm up & decay (incorrect turning point as seen from Tensorboard log) was fixed with commit 1f998b59d4e799b31f7c5b9ca4f642b727fd8882. Sorry for the confusion!

**Comment by madisonmay** [comment]
No worries!  Thank you for bringing that to our attention regardless, was certainly worth a check to make sure things were behaving properly.
